src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -111,12 +111,13 @@ class TaskReplaceActor(
   private def initialized: Receive = readinessBehavior orElse replaceBehavior
 
   def replaceBehavior: Receive = {
-    // New instance failed to start, restart it
-    case InstanceChanged(id, runSpecVersion, `pathId`, condition, Instance(instanceId, Some(agentInfo), state, tasksMap, runSpec, reservation)) if newInstanceIds(id) && considerTerminal(condition) =>
-      logger.warn(s"New instance $id is terminal on agent ${agentInfo.agentId} during app $pathId restart: $condition reservation: $reservation")
-      instanceTerminated(id)
-      instancesStarted -= 1
-      launchInstances().pipeTo(self)
+    // New instance task failed: instance goal is still Running so nothing to do here, the task will be rescheduled automatically
+    case InstanceChanged(id, runSpecVersion, `pathId`, condition, Instance(instanceId, Some(agentInfo), state, tasksMap, runSpec, reservation)) if newInstanceIds(id) && considerTerminal(condition) && state.goal == Goal.Running =>
+      logger.warn(s"New $id is terminal ($condition) on agent ${agentInfo.agentId} during app $pathId restart: $condition reservation: $reservation. Waiting for the task to restart...")
+
+    // New instance task failed: instance goal is *NOT* Running?! This should *NOT* happen and means, someone is interfering with current deployment!
+    case InstanceChanged(id, runSpecVersion, `pathId`, condition, Instance(instanceId, Some(agentInfo), state, tasksMap, runSpec, reservation)) if newInstanceIds(id) && considerTerminal(condition) && state.goal != Goal.Running =>
+      logger.error(s"New $id is terminal ($condition) on agent ${agentInfo.agentId} during app $pathId restart (reservation: $reservation) and the goal (${state.goal}) is *NOT* Running! This means that someone is interfering with current deployment!")
```

Comment:
This is more of a sanity check: basically, we don't expect anybody to interfere with the deployment (`MarathonSchedulerActor` lock should prevent this). Not sure what to do here since only one orchestrator logic should be responsible for setting instance goal and this behavior is considered a bug now. The deployment will be basically "stuck" (and has to be canceled/stopped by the user should this BUG surface)

Reply:
Why not nest this receive and the one above? It is hard to follow all `if`-statements after the extractor. What do you think about defining a new extractor?

```scala
case InstanceChanged(id, runSpecVersion, `pathId`, condition, Instance(instanceId, Some(agentInfo), State(Running, Terminal), tasksMap, runSpec, reservation)) if newInstanceIds(id)
```